src/site/controllers/challenge.js

```diff
@@ -61,21 +63,20 @@ exports.putNewChallenge = async (req, res, next) => {
       streamerId
     })
     const fee = configuration ? configuration.fee : DEFAULT_FEE
-
-    Challenge.add({
-      name: req.body.name,
-      description: req.body.description,
-      price: req.body.reward,
-      fee,
-      duration: req.body.duration,
-      nickname: req.body.nickname,
-      streamerId,
-      callback: (err, challengeId) => {
-        if (err) { return next(err) }
-
-        return res.redirect(`/challenge/${challengeId}/payment`)
-      }
-    })
+    new Challenge().create({
```

Comment:
it doesn't look really beneficial to have `ChallengeEntity` as a class if every time when you want to call the function you have to create in the memory new instance of the object. I'd say it better fits ChallengeEntity as a module.

Reply:
holywar

Reply:
no, every time creating an instance is just memory consuming... we can leave with this for now, that's sure, but when we will be creating something similar - better to choose another approach

Reply:
It 'll be remove by gc.

Reply:
using GC is not free

Reply:
??? I mean garbage collector

Reply:
me too